synapse/axon.py

```diff
@@ -187,25 +191,23 @@ def __init__(self, link, item):
         '''
         self.chunknum = 0
 
-class _AsyncQueue(s_coro.Fini):
+class _AsyncQueue(s_base.Base):
```

Comment:
Can we move this into `synapse.lib.queue`  and make sure he has dedicated unit tests??